lib/esp-scanning-lib.robot

```diff
@@ -0,0 +1,126 @@
+    ...    shim and grub files, ignored for windows type
+    ...    ${folder_name} - usually simply system name, ignored for windows
+    [Arguments]    ${type}    ${folder_name}=Microsoft
+
+    IF    "${type}"=="windows"
+        SSHLibrary.Execute Command    mkdir Microsoft
+        SSHLibrary.Execute Command    mkdir Microsoft/Boot
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O Microsoft/Boot/bootmgfw.efi
+    END
+    IF    "${type}"=="manager"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/elilo.efi
+    END
+    IF    "${type}"=="linux"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/shimx64.efi
+    END
+    SSHLibrary.Execute Command    wget ${GRUBX64_URL} -O ${folder_name}/grubx64.efi
+
+    # since switch to root user doesn't really wanna work
+    SSHLibrary.Execute Command    cp -r ~/${folder_name}/ /boot/efi/EFI/
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    sync
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    rm ${folder_name} -r
+
+Check Boot Menu For All Supported Systems
+    [Documentation]    This keyword scans the boot menu in search for all
+    ...    systems supported by the ESP scan feature. To be used after
+    ...    populating ESP with all the required files.
+    ...    Available modes:
+    ...    normal
+    ...    empty - keyword fails if the list is not empty
+    [Arguments]    ${mode}=normal
+
+    ${boot_list}=    Get Boot Menu Construction
+    Press Key N Times    1    ${ARROW_UP}
+    IF    "${mode}"!="empty"
+        ${boot_list}=    Get Setup Menu Construction    Debian
+    END
+    IF    "${mode}"=="empty"
+        Only N Occurrences    ${boot_list}    0    *Suse*
+        Only N Occurrences    ${boot_list}    0    *RedHat*
+    ELSE
+        Only N Occurrences    ${boot_list}    2    *Suse*
+        Only N Occurrences    ${boot_list}    2    *RedHat*
+    END
+    @{systems}=    Create List    Fedora    OpenSuse    Windows    CentOS    QubesOS
+    FOR    ${system}    IN    @{systems}
+        IF    "${mode}"=="empty"
+            Should Not Contain Match    ${boot_list}    *${system}*
+        END
+        IF    "${mode}"=="normal"
+            Should Contain Match    ${boot_list}    *${system}*
+        END
+        IF    "${mode}"=="double_entry_check"
+            Only N Occurrences    ${boot_list}    1    *${system}*
+        END
+    END
+
+Only N Occurrences
+    [Documentation]    This keyword ensures that there's Only N Occurrences
+    ...    of a given entry in a list
+    [Arguments]    ${list}    ${n}    ${entry}
+    ${x}=    Get Match Count    ${list}    ${entry}
+    Should Be Equal As Integers    ${x}    ${n}
+
+Remove All Supported Systems From Efi
+    Remove System From Efi    Fedora
```

Comment:
How about a list and iterate through that with a loop instead?

Reply:
Rewritten [here](https://github.com/Dasharo/open-source-firmware-validation/pull/31/files#diff-1c553d8d90529e6578e4c3b7bec1aea478c26edce6d99d353b8a35b02c05c67cR103)